components/devtools/actors/console.rs

```diff
@@ -89,6 +108,86 @@ pub struct ConsoleActor {
+        let input = msg.get("text").unwrap().as_str().unwrap().to_owned();
+        let (chan, port) = ipc::channel().unwrap();
+        self.script_chan
+            .send(DevtoolScriptControlMsg::EvaluateJS(
+                self.pipeline,
+                input.clone(),
+                chan,
+            ))
+            .unwrap();
+
+        //TODO: extract conversion into protocol module or some other useful place
+        let result = match port.recv().map_err(|_| ())? {
+            VoidValue => {
+                let mut m = Map::new();
+                m.insert("type".to_owned(), Value::String("undefined".to_owned()));
+                Value::Object(m)
+            },
+            NullValue => {
+                let mut m = Map::new();
+                m.insert("type".to_owned(), Value::String("null".to_owned()));
+                Value::Object(m)
+            },
+            BooleanValue(val) => Value::Bool(val),
+            NumberValue(val) => {
+                if val.is_nan() {
+                    let mut m = Map::new();
+                    m.insert("type".to_owned(), Value::String("NaN".to_owned()));
+                    Value::Object(m)
+                } else if val.is_infinite() {
+                    let mut m = Map::new();
+                    if val < 0. {
+                        m.insert("type".to_owned(), Value::String("-Infinity".to_owned()));
+                    } else {
+                        m.insert("type".to_owned(), Value::String("Infinity".to_owned()));
+                    }
+                    Value::Object(m)
+                } else if val == 0. && val.is_sign_negative() {
+                    let mut m = Map::new();
+                    m.insert("type".to_owned(), Value::String("-0".to_owned()));
+                    Value::Object(m)
+                } else {
+                    Value::Number(Number::from_f64(val).unwrap())
+                }
+            },
+            StringValue(s) => Value::String(s),
+            ActorValue { class, uuid } => {
+                //TODO: make initial ActorValue message include these properties?
+                let mut m = Map::new();
+                let actor = ObjectActor::new(registry, uuid);
+
+                m.insert("type".to_owned(), Value::String("object".to_owned()));
+                m.insert("class".to_owned(), Value::String(class));
+                m.insert("actor".to_owned(), Value::String(actor));
+                m.insert("extensible".to_owned(), Value::Bool(true));
+                m.insert("frozen".to_owned(), Value::Bool(false));
+                m.insert("sealed".to_owned(), Value::Bool(false));
+                Value::Object(m)
+            },
+        };
+
+        //TODO: catch and return exception values from JS evaluation
+        let reply = EvaluateJSReply {
+            from: self.name(),
+            input: input,
+            result: result,
+            timestamp: 0,
+            exception: Value::Null,
```

Comment:
Exceptions must be null if they are not present. We were sending an empty object over the wire, and devtools was interpreting it as an error.